system_metrics_collector/src/moving_average_statistics/moving_average.cpp

```diff
@@ -77,13 +77,16 @@ void MovingAverageStatistics::reset()
 void MovingAverageStatistics::addMeasurement(const double item)
 {
   std::lock_guard<std::mutex> guard(mutex);
-  count_++;
-  const double previous_average_ = average_;
-  average_ = previous_average_ + (item - previous_average_) / count_;
-  min_ = std::min(min_, item);
-  max_ = std::max(max_, item);
-  sum_of_square_diff_from_mean_ = sum_of_square_diff_from_mean_ + (item - previous_average_) *
-    (item - average_);
+
+  if (!std::isnan(item)) {
+    count_++;
+    const double previous_average_ = average_;
+    average_ = previous_average_ + (item - previous_average_) / count_;
+    min_ = std::min(min_, item);
```

Comment:
Something I recall from `rosbag2` for Windows support  for `std::min/max` is adding something like [this](https://github.com/ros2/rosbag2/blob/0a3ba417c017a9baa54c58e33f1d44296b8c1dec/rosbag2/CMakeLists.txt#L18) to your CMakeLists.txt